src/safeds_runner/server/pipeline_manager.py

```diff
@@ -80,21 +80,34 @@ def _handle_queue_messages(self) -> None:
         try:
             while self._messages_queue is not None:
                 message = self._messages_queue.get()
-                if self._websocket_target is not None:
-                    self._websocket_target.send(json.dumps(message.to_dict()))
+                message_encoded = json.dumps(message.to_dict())
+                # only send messages to the same connection once
+                for connection in set(self._websocket_target):
+                    connection.send(message_encoded)
         except BaseException as error:  # noqa: BLE001  # pragma: no cover
             logging.warning("Message queue terminated: %s", error.__repr__())  # pragma: no cover
 
     def set_new_websocket_target(self, websocket_connection: simple_websocket.Server) -> None:
```

Comment:
The `set_` prefix implies that something is overwritten, which is no longer the case. I'd generally suggest renaming this function `connect`...